internal/testutils/test_utils.go

```diff
@@ -28,3 +29,13 @@ func GenerateNormalizedJSON(j string) string {
 	n, _ := json.Marshal(i)
 	return string(n)
 }
+
+// SaveAsFormattedJSON save the object as a formatted JSON file to help
+// with investigations.
+func SaveAsFormattedJSON(file string, o interface{}) error {
```

Comment:
Why are we adding this function?

Reply:
Speedup diagnostic of the specific issue: this is just called in case of test failure, a diff between the files helps spot the issues.